telemetry/software/telemetry/library.py

```diff
@@ -0,0 +1,116 @@
+"""
+objective - grab data from serial, parse, pass into redis
+timeseries database in realtime
+"""
+
+import serial
+import time
+from redis import Redis
+from redistimeseries.client import Client
+
+
+# Data stream parent class with main pass_data() method call
+# 2-3 data stream children - serial stream, logfile stream, custom pass_data() methods
+class DataStream(object):
+    def __init__(self, data_channels):
+        # self.start_stream = True
+        self.data_channels = data_channels
+
+    # def close_stream(self):
+    #     self.start_stream = False
+
+    def return_data_channels(self):
+        return self.data_channels
+
+
+class ArduinoSerialIn(DataStream):
+    """
+    Class to represent arduino serial datastream object in for testing purposes
+    """
+
+    def __init__(
+        self, baudrate=9600, port_name="/dev/ttyUSB0", data_channels=["arduino_data"]
+    ):
+        DataStream.__init__(self, data_channels)
+        self.ser = serial.Serial(port_name, baudrate)
+
+    def read_line(self):
+        """
+        returns data float
+        """
+        flt = self.parse_line(self.ser.readline())
+        print(flt)  # for debug
```

Comment:
use a debug mode for this?

Reply:
By that I mean in `__init__` you define a `self.debug = False` and then check if debug is `True` to print out values like this

Reply:
should define it in the `config.py` file. and I think we should use `logging` instead of print

Reply:
Oof. Idk, like, we could use logging (its objectively more powerful), but imo it adds more development time than it saves. I'm personally fine with prints the way they are written, but if we want to use logging thats fine too.